helix-view/src/document.rs

```diff
@@ -20,6 +21,21 @@ pub enum Mode {
     Insert,
 }
 
+/// Represents one of the valid Unicode line endings.
+/// Also acts as an index into `LINE_ENDINGS`.
+#[derive(PartialEq, Copy, Clone, Debug)]
+pub enum LineEnding {
+    None = 0, // No line ending
+    Crlf = 1, // CarriageReturn followed by LineFeed
+    LF = 2,   // U+000A -- LineFeed
+    VT = 3,   // U+000B -- VerticalTab
+    FF = 4,   // U+000C -- FormFeed
+    CR = 5,   // U+000D -- CarriageReturn
+    Nel = 6,  // U+0085 -- NextLine
+    LS = 7,   // U+2028 -- Line Separator
+    PS = 8,   // U+2029 -- ParagraphSeparator
+}
+
```

Comment:
The second line of the doc comment here doesn't make sense here, since you didn't bring over the `LINE_ENDINGS` array (and I don't think you need to).  Probably best to delete that.

Also, in retrospect I don't know why I had a `None` variant in there.  That would probably be better represented by just wrapping `LineEnding` in an `Option` where needed.

Reply:
We can also omit the explicit numbering, it's not necessary since there's no more `LINE_ENDINGS` array